### PR TITLE
Pull Request for todo-testing-mock: To-do-List

### PR DESCRIPTION
**Pull request by anita00001**
In this branch, we have written the mock and test for local storage updates.

## Features:
- [ ] Use jest framework for testing.
- [ ] Create a test file(s) for files containing
       - a function for editing the task description.
       - a function for updating an item's 'completed' status.
       - the "Clear all completed" function.
- [ ] Use the mock storage to mock the local storage updates.

@david30-maker

### CODE REVIEW

File: src/edit.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  { description: 'Task1', completed: false, index: 1 },
+  { description: 'Update GitHub Profile', completed: false, index: 2 },
+  { description: 'Task3', completed: false, index: 3 },
+];
+
+test('Update ToDo item', () => {
+  expect(
+    editTask(previousTask, 2 - 1, 'Update GitHub Profile'),
+  ).toEqual(newTask);
+});
```
Comment (OyePriscilla) [inline src/edit.test.js]: - [x] According to the project requirement, please kindly group your test using the `describe()` method. **Optionally**, you can add one or more tests in order to bring out the essence of the `describe()` method. A describe block is used for organizing test cases in logical groups of tests. ℹ️ 

**> Note, I commend how you implemented the test cases for `localstorage.test.js`. 👍**

File: src/deleteAll.test.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+const remainingList = [
+  { description: 'Task2', completed: false, index: 1 },
+];
+
+test('Delete All Completed ToDo items', () => {
+  expect(
+    clearCompleted(oldList),
+  ).toEqual(remainingList);
+});
```
Comment (OyePriscilla) [inline src/deleteAll.test.js]: - [x] Please kindly group your test using the describe() method. **Optionally**, you can add one or more tests in order to bring out the essence of the describe() method. A describe block is used for organizing test cases in logical groups of tests. ℹ️

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @anita00001 and @david30-maker 👏 👏, 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Use jest framework for testing ✔️ 
Write unit tests for "editing", "clear all completed" and "completed status?" ✔️ 
All tests passed ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Zuhaib042 — APPROVED** [review]
Hi @anita00001 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
approve

## Highlights
✅ Good Implementation of suggested changes
✅ Professional PR

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**